xhistogram/test/fixtures.py

```diff
@@ -12,3 +15,33 @@ def raise_if_computed():
         a = a.rechunk(chunks)
 
     return a
+
+
+@pytest.fixture(scope="module")
+def dataarray_factory():
+    def _dataarray_factory(shape=(5, 20)):
+        data = np.random.randn(*shape)
+        dims = ["x", "y", "z", "w"][: len(shape)]
+        da = xr.DataArray(data, dims=dims, name="T")
+        return da
+
+    return _dataarray_factory
+
+
+@pytest.fixture(scope="module")
+def dataset_factory():
+    """Random dataset with every variable having the same shape"""
+
+    def _dataset_factory(ndim=2, n_vars=2):
+        shape = (8, 9, 10, 11)[:ndim]
+        dims = ["x", "y", "z", "w"][:ndim]
+        var_names = ["T", "S", "V", "U"]
```

Comment:
It looks like this may not behave as intended when `ndim > 4`. Could we generalize `dataset_factory` and `dataarray_factory` a bit to make sure they work for valid inputs like `ndim=5`, `shape=(3, 4, 5, 6, 7)`, etc.

For example, using things like `dims = [f"dim_{i}" for i in range(ndim)]` or `var_names = [uuid.uuid4().hex for _ in range(n_vars)]`